src/goose/brokers/redis/api/dead_jobs.clj

```diff
@@ -18,9 +18,7 @@
 
 (defn find-by-id [conn id]
   (let [
-        ;; TODO: Think about this limit more carefully. Why is it hardcoded?
-        ;; It used to be 1, which was breaking with the new scan implementation.
-        limit 10
+        limit 1
```

Comment:
This will cause breakage, because the limit applies before filtering, not after. Thus, it is possible for this function to return `nil` even if the given ID is actually present.

Reply:
Flipped the take & filter